data/projects/nested_languages.md

```diff
@@ -1,28 +1,41 @@
 Multiple programming languages can coexist in a single source file. coala would
 benefit from the ability to determine the programming language of each part of
 a file. If we have e.g. a PHP file, it is likely that we'd have chunks of html
-inside a few of them. We want some method of finding specific languages inside
-other languages and analyze them accordingly for e.g. awk inside bash, JSON
-inside Javascript, etc.
+inside a few of them. We would like coala to be able to still run meaningful
+analysis on those files.
+Other examples include:
+
+* awk inside bash
+* JSON inside Javascript
+* HTML, LaTeX or others inside Jinja2
 
 This project is about enabling coala to deal with those situations and allow
 people to write code analysis similar to how they already do it while being
 applicable to the right locations at the right files.
 
+There are several ways to approach this. The most straightforward would be to identifiy common combinations
+of languages and come up with a clever way to implement support for these combinations individually. A more
+advanced way would be to come up with a more abstract way to define how arbitrary combinations could be supported.
```

Comment:
You can replace 'abstract way' with 'abstract solution'. It fits better. Try to avoid word repetition in a single sentence.